### PR TITLE
drivers: espi: npcx: update for espi reset level

### LINKED ISSUE #68557: Handling of ESPI_BUS_RESET for NPCX driver doesn't match other eSPI drivers
The NPCX eSPI driver, when posting the `ESPI_BUS_RESET` event sets the `evt_data`  as follows
* 1 = eSPI reset asserted
* 0 = eSPI reset deasserted

This is the opposite of the other eSPI drivers (ITE and Microchip) which set `evt_data` to match the physical level of the eSPI reset.

### PR DESCRIPTION
**Pull request by TomChang19**
This CL updates the event data returned by eSPI reset ISR.  
The event data is indicated the state of eSPIRST.  
The 0 means eSPI bus in reset, and 1 means eSPI bus out-of-reset.

Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/68557

### CODE REVIEW

File: drivers/espi/espi_npcx.c [hunk 1]
```diff
@@ -611,11 +611,11 @@ static void espi_vw_espi_rst_isr(const struct device *dev, struct npcx_wui *wui)
 	struct espi_npcx_data *const data = dev->data;
 	struct espi_event evt = { ESPI_BUS_RESET, 0, 0 };
 
-	data->espi_rst_asserted = !IS_BIT_SET(inst->ESPISTS,
+	data->espi_rst_level = IS_BIT_SET(inst->ESPISTS,
 						NPCX_ESPISTS_ESPIRST_LVL);
```
Comment (fabiobaltieri) [inline drivers/espi/espi_npcx.c:615]: can you update the indentation?

Comment (TomChang19) [reply drivers/espi/espi_npcx.c]: Done.

**Review by fabiobaltieri — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Comment by keith-zephyr** [comment]
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/68557

**Review by MulinChao — APPROVED** [review]

**Comment by keith-zephyr** [comment]
Note that this fix conforms with the API contract defined by pending PR https://github.com/zephyrproject-rtos/zephyr/pull/68492.

**Pull request merged**